gateway/src/shard/processor/socket_forwarder.rs

```diff
@@ -41,18 +37,12 @@ impl SocketForwarder {
         #[cfg(feature = "tracing")]
         tracing::debug!("starting driving loop");
 
-        loop {
-            let timeout = sleep(Self::TIMEOUT).fuse();
-            tokio::pin!(timeout);
-
-            let rx = Box::pin(self.rx.recv().fuse());
-            let tx = Box::pin(self.stream.next().fuse());
+        let timeout = sleep(Self::TIMEOUT);
+        tokio::pin!(timeout);
```

Comment:
Should this not be moved into the loop since it is equivalent to `sleep_until(Instant::now() + duration)` and having it there will cause it to always fire after the timeout.